src/gtk/toga_gtk/widgets/table.py

```diff
@@ -1,80 +1,53 @@
 from gi.repository import Gtk
 
-from .base import Widget
+from .tree import Tree
 
 
-class Table(Widget):
-    def create(self):
-        self.store = Gtk.ListStore(*[object] + [str for h in self.interface.headings])
-
-        # Create a table view, and put it in a scroll view.
-        # The scroll view is the native, because it's the outer container.
-        self.treeview = Gtk.TreeView(model=self.store)
-        self.selection = self.treeview.get_selection()
-        self.selection.set_mode(Gtk.SelectionMode.SINGLE)
-        self.selection.connect("changed", self.on_select)
-
-        for i, heading in enumerate(self.interface.headings):
-            renderer = Gtk.CellRendererText()
-            column = Gtk.TreeViewColumn(heading, renderer, text=i + 1)
-            self.treeview.append_column(column)
-
-        self.native = Gtk.ScrolledWindow()
-        self.native.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.AUTOMATIC)
-        self.native.add(self.treeview)
-        # self.native.set_min_content_width(200)
-        # self.native.set_min_content_height(200)
-        self.native.interface = self.interface
+class Table(Tree):
 
     def on_select(self, selection):
-        if self.interface.on_select:
+        if hasattr(self.interface, "on_select") and self.interface.on_select:
```

Comment:
This shouldn't be required. The fix will generally be to ensure that `self._on_select` is primed with a value of None *before* the implementation is created.